vite.config.ts

```diff
@@ -0,0 +1,53 @@
+import { resolve } from "path";
+import { defineConfig, mergeConfig } from "vite";
+import { defineConfig as defineVitestConfig } from "vitest/config";
+import react from "@vitejs/plugin-react";
+
+const vitestConfig = defineVitestConfig({
+    test: {
+        globals: true,
+        include: ["test/**/*.{ts,tsx}"],
+    },
+});
+
+export default mergeConfig(
+    defineConfig({
+        plugins: [react()],
+        build: {
+            lib: {
+                // Could also be a dictionary or array of multiple entry points
+                entry: resolve(__dirname, "src/index.tsx"),
+                name: "PointCloud",
+                // the proper extensions will be added
+                fileName: "index",
+                formats: ["es", "cjs"],
+            },
+            minify: false,
+            rollupOptions: {
+                // make sure to externalize deps that shouldn't be bundled
+                // into your library
+                external: [
+                    "react",
```

Comment:
Weird that you need to do this explicitly...

Reply:
The default vite behavior is to bundle all deps with the library. I don't really like it either

Reply:
Removed extraneous config for umd modules as well